src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -1265,16 +1265,9 @@ def _validate_vm_vmss_msi(cmd, namespace, is_identity_assign=False):
 
     # For "az vm identity assign", "--scope" must be passed in when assigning a role to the managed identity
     if is_identity_assign:
-        role_is_explicitly_specified = getattr(namespace.identity_role, 'is_default', None) is None
-        if not namespace.identity_scope and role_is_explicitly_specified:
+        if not namespace.identity_scope and namespace.identity_role or not namespace.identity_role and namespace.identity_scope:
```

Comment:
Adding parentheses appropriately can increase readability 😉
```suggestion
        if (not namespace.identity_scope and namespace.identity_role) or (not namespace.identity_role and namespace.identity_scope):
```

Reply:
addressed :)